src/array.ts

```diff
@@ -3,6 +3,7 @@ import type { KeysOfType } from "./utils";
 declare global {
   interface Array<T> {
     unique(): Array<T>;
+    uniqueObjectsByKey(key: string): Array<T>;
```

Comment:
Fwiw can we rename this to uniqueByKey? I guess it can only work on objects but 🤷 I think the "by key" suffix implies that sufficiently.